tests/sdk/queries/test_query_filter.py

```diff
@@ -414,3 +419,112 @@ def test_filter_group_when_changed_filter_clause_has_correct_json_representation
         '{"operator":"IS", "term":"term", "value":"value2"},'
         '{"operator":"IS", "term":"term", "value":"value3"}]}'
     )
+
+
+class TestQueryFilterTimestampField:
+    @pytest.mark.parametrize(
+        "timestamp",
+        [
+            (TEST_TIMESTAMP),
+            (1599736333.0),
+            (1599736333),
+            datetime.strptime(TEST_TIMESTAMP, "%Y-%m-%d %H:%M:%S"),
+        ],
+    )
+    def test_on_or_after(self, timestamp):
+        filter_query = {
+            "filterClause": "AND",
+            "filters": [
+                {
+                    "operator": "ON_OR_AFTER",
+                    "term": "override_timestamp_field_name",
+                    "value": "2020-09-10T11:12:13.000Z",
+                }
+            ],
+        }
+
+        qf = QueryFilterTimestampField.on_or_after(timestamp)
+        assert dict(qf) == filter_query
+
+    @pytest.mark.parametrize(
+        "timestamp",
+        [
+            (TEST_TIMESTAMP),
+            (1599736333.0),
+            (1599736333),
+            datetime.strptime(TEST_TIMESTAMP, "%Y-%m-%d %H:%M:%S"),
+        ],
+    )
+    def test_on_or_before(self, timestamp):
+        filter_query = {
+            "filterClause": "AND",
+            "filters": [
+                {
+                    "operator": "ON_OR_BEFORE",
+                    "term": "override_timestamp_field_name",
+                    "value": "2020-09-10T11:12:13.000Z",
+                }
+            ],
+        }
+        assert dict(QueryFilterTimestampField.on_or_before(timestamp)) == filter_query
+
+    @pytest.mark.parametrize(
```

Comment:
These are good - one other test we could also do is where there are digits after the decimal (not just `.0`) and make sure the timestamp works (doesn't end in `000Z`).

Reply:
Added tests.